Assets/Scripts/Singletons/ShipsManager.cs

```diff
@@ -0,0 +1,32 @@
+﻿using UnityEngine;
+
+public class ShipsManager : MonoBehaviour
+{
+    public static ShipsManager Instance;
+
+    public Ship[] ships;
+
+    private void Awake()
+    {
+        if (FindObjectsOfType(GetType()).Length > 1)
+        {
+            Destroy(gameObject);
+        }
+
+        if (Instance == null)
+        {
+            Instance = this;
+        }
+        else if (Instance == this)
+        {
+            Destroy(Instance.gameObject);
+            Instance = this;
+        }
+        DontDestroyOnLoad(this.gameObject);
+    }
+
+    public static void DamageShip(int shipId, int damage)
+    {
+        ships[shipId].hullIntegrity = Mathf.Max(0, ships[shipId].hullIntegrity - damage);
```

Comment:
Id should be index to make it abundantly clear to everyone how it works. We might wanna find a better way to find ships by id, that way if the scriptable object get out of order, we don't run into issues. This will make it much easier to add things in the future (and prevent rework)

Reply:
While it might be a hair slower, you could just do a for loop through the ships and check the IDs, rather than try to line up the indexes

Reply:
`int index = System.Array.IndexOf(PilotsManager.Instance.pilots, mission.pilot);
PilotsManager.Instance.pilots[index].xp += Random.Range(xpMin, xpMax);
`

This was commented out in the xp file. Just out of interest:
- Could we do this instead?
- Is it more performant?

I'm guessing it might fuck things up if a scriptable object was duplicated in the editor.

Reply:
On that note, if they are duplicated in the editor, there will be duplicate ids, so this will update all the scriptable objects matching the id passed in.

Reply:
Either way, I like your way better.